controllers/openstackconfiggenerator_controller.go

```diff
@@ -683,10 +686,32 @@ func (r *OpenStackConfigGeneratorReconciler) verifyNodeResourceStatus(instance *
 			return msg, false, nil
 		}
 	}
+	hasControllersForRoles := r.isControllerGenerator(vmset, bmset)
 
 	return msg, true, nil
 }
 
+//
+// Is this ConfigGenerator creating resources for a Controller
+//
```

Comment:
This part is still in progress and perhaps I should break it out into a separate patch. The intent is to use this function to only execute the overcloud.json harvesting if the stack *is* the ControlPlane (i.e. has Controller roles in it)